### PR TITLE
[NetAppFiles] Release netapp microsoft.net app 2024-07-01

### PR DESCRIPTION
**Pull request by audunn**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by audunn** [comment]
EffectiveNetworkFeatures approved here https://github.com/Azure/azure-rest-api-specs-pr/pull/18325 

Service level Flexible approved here https://github.com/Azure/azure-rest-api-specs-pr/pull/18494 and  renamed https://github.com/Azure/azure-rest-api-specs-pr/pull/18733

Also included change in upcoming (to be merged) and approved api-version 2024-05-01  https://github.com/Azure/azure-rest-api-specs/pull/30299

**Comment by audunn** [comment]
Breaking change https://msazure.visualstudio.com/One/_workitems/edit/29422166

**Comment by ms-bogdan-rotaru** [comment]
>         "backupPolicyId": {

Can you please indicate why this is not marked as "arm-id" anymore?

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2024-07-01/netapp.json:9298 in d26c18e. [](commit_id = d26c18e36e859d75649bb6b987ae30ae90ce8f78, deletion_comment = False)

**Comment by ms-bogdan-rotaru** [comment]
>         "backupVaultId": {

Can you please indicate why this is not marked as "arm-id" anymore?

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2024-07-01/netapp.json:9306 in d26c18e. [](commit_id = d26c18e36e859d75649bb6b987ae30ae90ce8f78, deletion_comment = False)

**Comment by ms-bogdan-rotaru** [comment]
>         },

This is inconsistent with line 8095 where "backupPolicyId" is not defined as "arm-id".

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2024-07-01/netapp.json:9333 in c01e40b. [](commit_id = c01e40b279b17f926adeaa5c075e024e15efe5f1, deletion_comment = False)

**Comment by audunn** [comment]
> > ```
> >     },
> > ```
> 
> This is inconsistent with line 8095 where "backupPolicyId" is not defined as "arm-id".
> 
> Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2024-07-01/netapp.json:9333 in [c01e40b](https://github.com/Azure/azure-rest-api-specs/commit/c01e40b279b17f926adeaa5c075e024e15efe5f1). [](commit_id = [c01e40b](https://github.com/Azure/azure-rest-api-specs/commit/c01e40b279b17f926adeaa5c075e024e15efe5f1), deletion_comment = False)

Good catch thank you very much.
This is indeed not the right property. The backupPolicyId in volumeBackupProperties should be changed to arm-id.  
Also added arm-id on backupsMigrationRequest backupVaultId that should be arm-id as well.

PR Updated with fixes based on comments.

**Comment by ms-bogdan-rotaru** [comment]
>           "description": "Backup Policy GUID ID"

Is this a GUID or a resource identifier?

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2024-07-01/netapp.json:8098 in c5efbd8. [](commit_id = c5efbd80bab536b0bfb1e424f865cd5bdf089357, deletion_comment = False)

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2024-07-01/netapp.json [hunk 1]
```diff
@@ -6588,7 +6588,7 @@
           "title": "usageThreshold",
           "type": "integer",
           "format": "int64",
-          "description": "Maximum storage quota allowed for a file system in bytes. This is a soft quota used for alerting only. For regular volumes, valid values are in the range 50GiB to 100TiB. For large volumes, valid values are in the range 100TiB to 500TiB, and on an exceptional basis, from to 2400GiB to 2400TiB. Values expressed in bytes as multiples of 1 GiB.",
+          "description": "Maximum storage quota allowed for a file system in bytes. This is a soft quota used for alerting only. Minimum size is 100 GiB. Upper limit is 100TiB, 1 PiB for LargeVolume or 2400Tib for LargeVolume on exceptional basis. Specified in bytes.",
           "minimum": 53687091200,
```
Comment (einarnetapp) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2024-07-01/netapp.json:6592]: Updated description does not match minimum limit

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2024-07-01/netapp.json [hunk 2]
```diff
@@ -7411,7 +7435,7 @@
           "title": "usageThreshold",
           "type": "integer",
           "format": "int64",
-          "description": "Maximum storage quota allowed for a file system in bytes. This is a soft quota used for alerting only. For regular volumes, valid values are in the range 50GiB to 100TiB. For large volumes, valid values are in the range 100TiB to 500TiB, and on an exceptional basis, from to 2400GiB to 2400TiB. Values expressed in bytes as multiples of 1 GiB.",
+          "description": "Maximum storage quota allowed for a file system in bytes. This is a soft quota used for alerting only. For regular volumes, valid values are in the range 50GiB to 100TiB. For large volumes, valid values are in the range 100TiB to 1PiB, and on an exceptional basis, from to 2400GiB to 2400TiB. Values expressed in bytes as multiples of 1 GiB.",
```
Comment (einarnetapp) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2024-07-01/netapp.json:7438]: Should this not be in sync with other usageThreshold description?

**Review by einarnetapp — APPROVED** [review]

**Pull request merged**